scipy/special/meson.build

```diff
@@ -392,7 +404,6 @@ python_sources = [
   'cython_special.pyi',
   'orthogonal.py',
   'sf_error.py',
-  'specfun.py',
```

Comment:
I'm not sure if we can remove this file, it's part of the public API and is not announced as being removed until SciPy 2.0.0

Reply:
Ah, thanks for stepping in. @steppi Thought we might be able to?

Basically I just need someone to tell me what's up. CI started complaining again specfun.py

Reply:
Right, so I reverted all specfun related changes. Essentially we're just fighting the linter.

The issue was here: https://github.com/scipy/scipy/actions/runs/8621413202/job/23630196758?pr=20320

I know you told me earlier I could delete names from `__all__` that weren't being used, but now it complains about literally every name in `__all__`. Should I just make `__all__` empty?

Reply:
This is how the file should be 
```
diff --git a/scipy/_lib/pocketfft b/scipy/_lib/pocketfft
index 936714274..0bf2b5122 160000
--- a/scipy/_lib/pocketfft
+++ b/scipy/_lib/pocketfft
@@ -1 +1 @@
-Subproject commit 9367142748fcc9696a1c9e5a99b76ed9897c9daa
+Subproject commit 0bf2b51224e7c4e973724bac67d2577c7831e27c
diff --git a/scipy/special/specfun.py b/scipy/special/specfun.py
index 28ac76d07..1d3d0fd2c 100644
--- a/scipy/special/specfun.py
+++ b/scipy/special/specfun.py
@@ -5,6 +5,11 @@
 from scipy._lib.deprecation import _sub_module_deprecation
 
 __all__ = [  # noqa: F822
+    'clpmn',
+    'lpmn',
+    'lpn',
+    'lqmn',
+    'pbdv'
 ]
 
 
@@ -14,5 +19,5 @@ def __dir__():
 
 def __getattr__(name):
     return _sub_module_deprecation(sub_package="special", module="specfun",
-                                   private_modules=["_specfun"], all=__all__,
+                                   private_modules=["_basic"], all=__all__,
                                    attribute=name)
```
This appears to be a bug in ruff see #10095